cmd/collector/app/zipkin/http_handler_test.go

```diff
@@ -72,39 +71,13 @@ func initializeTestServer(err error) (*httptest.Server, *APIHandler) {
 }
 
 func TestViaClient(t *testing.T) {
-	server, handler := initializeTestServer(nil)
+	server, _ := initializeTestServer(nil)
 	defer server.Close()
 
-	zipkinSender, err := zipkinTransport.NewHTTPTransport(
-		server.URL+`/api/v1/spans`,
-		zipkinTransport.HTTPBatchSize(1),
-	)
+	zexp, err := zipkin.New(server.URL+`/api/v1/spans`, zipkin.WithClient(server.Client()))
 	require.NoError(t, err)
-
-	tracer, closer := jaegerClient.NewTracer(
-		"test",
-		jaegerClient.NewConstSampler(true),
-		jaegerClient.NewRemoteReporter(zipkinSender),
-	)
-	defer closer.Close()
-
-	tracer.StartSpan("root").Finish()
-
-	waitForSpans(t, handler.zipkinSpansHandler.(*mockZipkinHandler), 1)
-}
-
-func waitForSpans(t *testing.T, handler *mockZipkinHandler, expecting int) {
-	assert.Eventuallyf(
-		t,
-		func() bool {
-			return len(handler.getSpans()) == expecting
-		},
-		2*time.Second,
-		time.Millisecond,
-		"expecting to receive %d span(s), have %d span(s)",
-		expecting,
-		len(handler.getSpans()),
-	)
+	assert.NoError(t, zexp.Shutdown(context.Background()))
+	assert.NoError(t, zexp.ExportSpans(context.Background(), nil))
```

Comment:
why do you think we had `waitForSpan` function?

Reply:
So you don't see a difference between _exporting_ spans and _receiving_ them? What is this whole file is supposed to test?

Reply:
If I misunderstood here, would I just need to rmv it from that TestViaClient()? Since L139 requires this.

Reply:
This is what the test was doing before your change:
  * create HTTP server/handler receiving Zipkin span
  * use Jaeger SDK with Zipkin exporter to generate spans and export them to that server/handler
  * validate that the handler received the expected spans

Reply:
Thanks Yuri for this info. So I removed waitForSpans(...) from TestViaClient() to get it not checked for spans received. I wondered if we need to init otel tracer for it with Zipkin exporter to generate spans. If yes, how?

Reply:
> so I removed waitForSpans(...) from TestViaClient() to get it not checked for spans received. 

so you still this it was the right thing to do to remove it?

> we need to init otel tracer for it with Zipkin exporter to generate spans. If yes, how?

why is this different from all the other times you already initialized OTEL tracer?

Reply:
So now I got you. That had not to be removed against the spans generated with some purpose.
> why is this different from all the other times you already initialized OTEL tracer?

I think becaz otel doesn't support for this `NewRemoteReporter(zipkinSender)`.

Reply:
> I think becaz otel doesn't support for this NewRemoteReporter(zipkinSender).

you previously instantiated tracer with OTLP exporter. Why is Zipkin exporter any different?

Reply:
@yurishkuro I am still getting no spans after init tracer. LMK if I am missing something.

Reply:
You need to call waitForSpans, export happens async.

Reply:
When I call it, zipkin exporter doesn't reflect exported spans. I mean `len(handler.zipkinSpansHandler.(*mockZipkinHandler).spans)` remains 0.

Reply:
Did you debug it? Is the exporter successful in exporting data? (I' pretty sure about what the issue is, but you should be able to troubleshoot issues like this yourself, especially in this very narrow setup).

Reply:
Yes Yuri, that's what I have been doing debugging the test so far. Let me take a look again after many attempts. Thanks.

Reply:
When the client can't connect to the server, one of the first things to check if the client has the right address. But also, client should be able to provide / log an error message indicating why it cannot send data.

Reply:
Seems to be listening on different port.

Reply:
Port is provided in server.URL, it should be correct. Anything else?

<img width="603" alt="image" src="https://github.com/jaegertracing/jaeger/assets/3523016/0a2d6957-c59c-48ed-ba17-41b33f8849f9">

Reply:
Sorry Yuri, I have changed that code a bit locally. After debugging, I am stuck.
![image](https://github.com/jaegertracing/jaeger/assets/94980910/5f561b0f-af7f-412e-8b56-d290b23cf98e)
 LMK!

Reply:
see my previous comment. You changed the URL for v2.

Reply:
Yes I know I did it intentionally to figure out the zipkin exporter. But still I don't get despite `/api/v1/spans`